utils/python/CIME/buildlib.py

```diff
@@ -0,0 +1,49 @@
+from CIME.XML.standard_module_setup import *
+from CIME.case import Case
+"""
+buildlibx - given the caseroot and component name(cmp), build the stub component
+"""
+
+class buildlib(object):
+    def __init__(self, case, model, comp):
+        self.case = case
+        self.model = model
+        self.comp = comp
+        self.caseroot = case.get_value("CASEROOT")
```

Comment:
I think Jim's suggestion makes sense - because different component sets (like the CESM prognostic components have more variables that are needed than the dead components. That brings up an interesting issue - where does having buildlib.py make sense. The buildlib for cam is not quite the same as the buildlib for clm and is different than the buildlib for pop. I'm really not sure it makes sense to introduce new methods for each of these in the buildlib.py - and am wondering if we shouldn't just have each prognostic component have its own separate buildlib?  In that case, buildlib.py would only be used for CIME components - is that the consensus?